runtime/lua/vim/shared.lua

```diff
@@ -801,7 +801,7 @@ end
 ---@return string String with whitespace removed from its beginning and end
 function vim.trim(s)
   vim.validate('s', s, 'string')
-  return s:match('^%s*(.*%S)') or ''
+  return s:gsub('^%s+', ''):match('^.*%S') or ''
```

Comment:
Can you add a comment here explaining why we don't use just `match`. This I'll prevent anyone changing this in the future.

Reply:
Done.